content/documentation/pages/6-recipes/7-scaling/2-autoscaling.md

```diff
@@ -0,0 +1,181 @@
+  kubectl patch cm my-release-prometheus-server --patch "$(cat ./kubernetes/helm/prometheus/prometheus-configmap.yaml)"
+  ```
+
+<!--END_TABS-->
+
+Register latest [kafka-docker](https://dataflow.spring.io/kafka-docker-latest) app starters:
+
+```shell
+app import --uri https://dataflow.spring.io/kafka-docker-latest --force
+```
+
+## Autoscaling Recipe
+
+### Create stream pipeline
+
+```shell
+stream create --name scaletest --definition "time --fixed-delay=995 --time-unit=MILLISECONDS | transform --expression=\"payload + '-' + T(java.lang.Math).exp(700)\" | log"
+```
+
+The `time` source generates current timestamp messages on a fixed time-interval (995ms = ~1msg/s), the `transform` processor performs a math operation to simulate a high CPU processing, and the `log` sink prints the transformed message payload.
+
+### Deploy pipeline with data partitioning
+
+```shell
+stream deploy --name scaletest --properties "app.time.producer.partitionKeyExpression=payload,app.transform.spring.cloud.stream.kafka.binder.autoAddPartitions=true,app.transform.spring.cloud.stream.kafka.binder.minPartitionCount=4"
+```
+
+The `producer.partitionKeyExpression=payload` property configures time’s output binding for partitioning. The partition key expression uses the message payload (e.g. the toString() value of the current timestamp) to compute how the data needs to be partitioned to the downstream output channels.
+The `spring.cloud.stream.kafka.binder.autoAddPartitions` deployment property instructs the Kafka binder to create new partitions when required. This is required if the topic is not already over-partitioned.
+The `spring.cloud.stream.kafka.binder.minPartitionCount` property sets the minimum number of partitions that the Kafka binder configures on the topic, which is where the transform-processor is subscribing for new data.
+
+<!--TIP-->
+
+When using RabbitMQ, the `app.log.spring.cloud.stream.kafka.binder.` deployment properties are not needed.
+
+<!--END_TIP-->
+
+![SCDF autoscaling stream deploying](images/scdf-autoscaling-stream-deploying.png)
+
+Once deployed with the help of the SCDF built in application Grafana dashboard to check the [stream application throughput metrics](%currentPath%/feature-guides/streams/monitoring/#prometheus-1):
+![SCDF autoscaling initial metrics](images/scdf-autoscaling-initial-metrics.png)
+
+The `time`, `transform` and `log` applications maintain the same message throughput (~ 1 message/s). The `transform` copes fine with the present load.
+
+### Increase pipeline load
+
+Now, let's increase the load by increasing the time-source's message production rate. By changing time's `time-unit` property from `MILLISECONDS` to `MICROSECONDS` the input rate will increase from one to a couple of thousands of messages per second.
+Note that the [stream update](%currentPath%/stream-developer-guides/continuous-delivery/) functionality allows to update the time application without stopping the entire stream:
```

Comment:
They are identical?

Reply:
There were too many changes and commentary, and back and forth, so it is likely that I misinterpreted it.

Reply:
While reviewing the PR in entirety, I see "rolling-update" and a few other subtle changes are missing here. Let's compare to make sure the text changes are consistent wherever they are reused.

Reply:
Do you mean we should use "rolling-update" term when explaining the time update?

Reply:
Correct .. its the same change I applied on the similar docs above. Because it is the same content, let's make sure they have identical use of terminology.